db.go

```diff
@@ -351,8 +351,15 @@ func (db *DB) init() error {
 // Close releases all database resources.
 // All transactions must be closed before closing the database.
 func (db *DB) Close() error {
+	db.rwlock.Lock()
+	defer db.rwlock.Unlock()
+
 	db.metalock.Lock()
 	defer db.metalock.Unlock()
+
+	db.mmaplock.RLock()
+	defer db.mmaplock.RUnlock()
+
```

Comment:
This locking order is consistent with the rest of the library (rwlock, metalock, mmaplock). The `mmaplock` takes a read lock because the `rwlock` prevents the write lock on `mmaplock`.